src/domain/users/users.repository.spec.ts

```diff
@@ -63,4 +72,231 @@ describe('UsersRepository', () => {
+        .build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress,
+        }),
+      ).rejects.toThrow(
+        new ConflictException('Cannot remove the current wallet'),
+      );
+    });
+
+    it('should throw a BadRequestException if there is only one wallet', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      const mockWallets = [walletBuilder().build()];
+      const mockUser = userBuilder().with('wallets', mockWallets).build();
+      mockUserRepository.findOne.mockResolvedValueOnce(mockUser);
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress: addressToRemove,
+        }),
+      ).rejects.toThrow(BadRequestException);
+    });
+
+    it('should throw a NotFoundException if the user is not found', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      mockWalletRepository.findOne.mockResolvedValueOnce(null);
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress: addressToRemove,
+        }),
+      ).rejects.toThrow(new NotFoundException('User not found'));
+    });
+
+    it('should throw a NotFoundException if the wallet does not exist for the user', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      const mockWallets = [walletBuilder().build(), walletBuilder().build()];
+      const mockUser = userBuilder().with('wallets', mockWallets).build();
+      mockUserRepository.findOne.mockResolvedValueOnce(mockUser);
+      mockWalletRepository.delete.mockResolvedValue({ affected: 0, raw: {} });
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress: addressToRemove,
+        }),
+      ).rejects.toThrow(
+        new NotFoundException(
+          `A wallet with address ${addressToRemove} does not exist for the current user`,
+        ),
+      );
+    });
+  });
+
+  describe('getUser', () => {
+    it('should return user information and associated wallets', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      const mockUser = userBuilder().with('id', 1).build();
+
+      const mockAuthenticatedWallet = walletBuilder()
+        .with('user', mockUser)
+        .with('address', authPayload.signer_address!)
```

Comment:
Nit: if we pass an address to `authPayloadDtoBuilder`, we can use that directly and not need to use a bang.